tests/integration/vpp/020_routes_test.go

```diff
@@ -63,3 +64,329 @@ func TestRoutes(t *testing.T) {
 		t.Fatalf("expected dump to contain both IPv4 and IPv6 routes")
 	}
 }
+
+func TestCRUDIPv4Route(t *testing.T) {
+	ctx := setupVPP(t)
+	defer ctx.teardownVPP()
+
+	ih := ifplugin_vppcalls.CompatibleInterfaceVppHandler(ctx.vppBinapi, logrus.NewLogger("test"))
+	ifName := "loop1"
+	ifIdx, err := ih.AddLoopbackInterface(ifName)
+	if err != nil {
+		t.Fatalf("creating interface failed: %v", err)
+	}
+	t.Logf("interface created")
+
+	ifIndexes := ifaceidx.NewIfaceIndex(logrus.NewLogger("test-iface1"), "test-iface1")
+	ifIndexes.Put("loop1", &ifaceidx.IfaceMetadata{
+		SwIfIndex: ifIdx,
+	})
+
+	//vrfMetaIdx := &vrfidx.VRFMetadata{Index: 0}
+	vrfIndexes := vrfidx.NewVRFIndex(logrus.NewLogger("test-vrf"), "test-vrf")
+	vrfIndexes.Put("vrf1-ipv4-vrf0", &vrfidx.VRFMetadata{Index: 0, Protocol: vpp_l3.VrfTable_IPV4})
+	vrfIndexes.Put("vrf1-ipv4-vrf2", &vrfidx.VRFMetadata{Index: 2, Protocol: vpp_l3.VrfTable_IPV4})
```

Comment:
Don't add the index for VRF 2 before it's successfully created. The VRF 0 is always exists.